test/unit/test_uop_symbolic.py

```diff
@@ -754,5 +755,109 @@ def test_unrolled_arange(self):
+
+        # Test basic overflow cases
+        idx = make_int64(0x7fffffff) + make_int64(1)  # Overflows int32
+        buf = UOp(Ops.DEFINE_GLOBAL, dtypes.float32.ptr(), arg=0x100000000)
+        load = UOp.load(buf.index(idx))
+        simplified = load.simplify()
+        assert simplified.src[0].src[1].dtype == dtypes.int64
+
+        # Test arithmetic operations with large numbers
+        idx2 = idx * make_int64(2)  # Large multiplication
+        load2 = UOp.load(buf.index(idx2))
+        simplified2 = load2.simplify()
+        assert simplified2.src[0].src[1].dtype == dtypes.int64
+
+        # Test complex indexing expressions
+        base_idx = make_int64(0x40000000)
+        offset = make_int64(0x40000000)
+        complex_idx = base_idx + offset  # Should need int64
+        load3 = UOp.load(buf.index(complex_idx))
+        simplified3 = load3.simplify()
+        assert simplified3.src[0].src[1].dtype == dtypes.int64
+
+        # Test negative indices
+        neg_idx = make_int64(-0x80000000) - make_int64(1)
+        load4 = UOp.load(buf.index(neg_idx))
+        simplified4 = load4.simplify()
+        assert simplified4.src[0].src[1].dtype == dtypes.int64
+
+        # Test mixed operations
+        mixed_idx = (base_idx * make_int64(2)) + make_int64(42)
+        load5 = UOp.load(buf.index(mixed_idx))
+        simplified5 = load5.simplify()
+        assert simplified5.src[0].src[1].dtype == dtypes.int64
+
+        # Test that small indices stay as int32
+        small_idx = UOp.const(dtypes.int32, 100) + UOp.const(dtypes.int32, 42)
+        load6 = UOp.load(buf.index(small_idx))
+        simplified6 = load6.simplify()
+        assert simplified6.src[0].src[1].dtype == dtypes.int32
+
+        # Test store operations
+        store1 = UOp.store(buf.index(idx), UOp.const(dtypes.float32, 1.0))
+        simplified_store = store1.simplify()
+        assert simplified_store.src[0].src[1].dtype == dtypes.int64
+
+        # Test chained operations
+        chain_idx = ((base_idx + offset) * make_int64(2)) // make_int64(3)
+        load7 = UOp.load(buf.index(chain_idx))
+        simplified7 = load7.simplify()
+        assert simplified7.src[0].src[1].dtype == dtypes.int64
+
+        # Test boundary conditions
+        boundary_idx = UOp.const(dtypes.int32, 0x7fffffff)  # Max int32
+        load8 = UOp.load(buf.index(boundary_idx))
+        simplified8 = load8.simplify()
+        assert simplified8.src[0].src[1].dtype == dtypes.int32  # Should still be int32
+
+        # Test ALU operations
+        alu_idx = (idx + make_int64(1)) * make_int64(2)
+        load9 = UOp.load(buf.index(alu_idx))
+        simplified9 = load9.simplify()
+        assert simplified9.src[0].src[1].dtype == dtypes.int64
+
+    def test_int64_conversion_from_int32(self):
+        # Create int32 indices that will overflow
+        idx = UOp.const(dtypes.int32, 0x7fffffff)  # Max int32
+        idx._min_max = (0x7fffffff, 0x7fffffff)  # Set min/max values
```

Comment:
For values like these, why not use dtype min/max function? Also the testing structure doesn't seem to systematically test all edge points. 0 indexing, -1 and mixing pos and neg tend to be some common ones.

Reply:
Good point. Now a lot of additional tests should be added.